Lookup with STS

baptistamarcelo opened the pull request:
Full example of parameter lookup using temporary credentials granted by STS AssumeRole.

##### SUMMARY
<!--- Describe the change below, including rationale and design decisions -->

<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->

##### ISSUE TYPE
<!--- Pick one below and delete the rest -->
- Docs Pull Request

##### COMPONENT NAME
<!--- Write the short name of the module, plugin, task or feature below -->
aws_ssm
##### ADDITIONAL INFORMATION
<!--- Include additional information to help people understand the change here -->
<!--- A step-by-step reproduction of the problem is helpful if there is no related issue -->

<!--- Paste verbatim command output below, e.g. before and after your change -->
```paste below

```

baptistamarcelo committed:
Lookup with STS

Full example of parameter lookup using temporary credentials granted by STS AssumeRole.

acozine commented:
Thanks @baptistamarcelo for adding examples to the Ansible documentation. There are three failures on your PR:

ERROR: lib/ansible/plugins/lookup/aws_ssm.py:102:1: W293 blank line contains whitespace (100%)
ERROR: lib/ansible/plugins/lookup/aws_ssm.py:115:161: E501 line too long (170 > 160 characters) (100%)
ERROR: lib/ansible/plugins/lookup/aws_ssm.py:120:1: W293 blank line contains whitespace (100%)

can you fix those?

mattclay commented:
bot_status

baptistamarcelo committed:
fixing PEP8

Fix for the following errors:

ERROR: lib/ansible/plugins/lookup/aws_ssm.py:102:1: W293 blank line contains whitespace (100%)
ERROR: lib/ansible/plugins/lookup/aws_ssm.py:115:161: E501 line too long (170 > 160 characters) (100%)
ERROR: lib/ansible/plugins/lookup/aws_ssm.py:120:1: W293 blank line contains whitespace (100%)

s-hertel commented on lib/ansible/plugins/lookup/aws_ssm.py:
```diff
@@ -100,6 +100,24 @@
+    sts_assume_role:
+      role_arn: 'arn:aws:iam::<acc_id>:role/<role_name>'
+      role_session_name: "{{ aws_sts_session_name | default('ansible-session') }}"
+      region: '<region_name>'
```
In case people copy and paste this, either make '<region_name>' a variable or replace the string with a valid one. Below too. But I think creating temporary credentials for plugins/modules should be documented somewhere more central since this isn't specific to this plugin.

Also, since this plugin uses boto3 a nicer way to do this is use a profile in your config file (usually found at ~/.aws/config) that has a source profile and the role to assume. For example:
```
# In ~/.aws/credentials:
[development]
aws_access_key_id=foo
aws_access_key_id=bar

# In ~/.aws/config
[profile crossaccount]
role_arn=arn:aws:iam:...
source_profile=development
```
and then use "{{ lookup('aws_ssm', 'my-parameter', aws_profile='crossaccount'}}" to assume the role instead of needing this first setup task.

samccann commented:
@baptistamarcelo can you please address the comments above in this PR?  Thanks!
